internal/postgres/reconcile_test.go

```diff
@@ -542,3 +543,187 @@ supplementalGroups:
 		assert.Assert(t, PodSecurityContext(cluster).SupplementalGroups == nil)
 	})
 }
+
+func TestUpgradeJob(t *testing.T) {
+	cluster := &v1beta1.PostgresCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testcluster",
+			Namespace: "testnamespace",
+		},
+	}
+	cluster.Spec.ImagePullPolicy = corev1.PullAlways
+	cluster.Spec.PostgresVersion = 12
```

Comment:
For this test, the values don't matter that much as long as the 'from' value is less than the version we're going to. With that said, I think this is ok to leave as is because we really shouldn't need to adjust it in the future.